lib/sycamore/sycamore/transforms/extract_graph.py

```diff
@@ -154,10 +185,121 @@ def extract_metadata(self, doc: Document) -> Document:
         return doc
 
 
-class ExtractMetadata(Map):
+class EntityExtractor(GraphExtractor):
+    """
+    Extracts entities chosen by the user by using LLMs
+
+    Args:
+        entities: A list of GraphEntity that determines what entities are extracted
+        llm: The LLM that is used to extract the entities
+    """
+
+    def __init__(self, entities: list[GraphEntity], llm):
+        self.entities = entities
+        self.llm = llm
+
+    def extract(self, docset: "DocSet") -> "DocSet":
+        """
+        Extracts entities from documents then creates a document in the docset where they are stored as nodes
+        """
+        docset.plan = ExtractFeatures(docset.plan, self)
+        docset = self.resolve(docset)
+        return docset
+
+    def _extract(self, doc: HierarchicalDocument) -> HierarchicalDocument:
+        from multiprocessing import Pool
```

Comment:
I'm a little worried about doing this in here. It's unclear that an EntityExtractor can internal spawn threads with its own configurations. Maybe calling it MultithreadedEntityExtractor or something will help?

@alexaryn @eric-anderson I'm curious about your thoughts here since you've played with this stuff in other parts

Reply:
Do you think adding a multithreading flag to entity extractor would be a good solution here?